subsys/bluetooth/services/cts.c

```diff
@@ -7,9 +7,16 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+
+#ifdef CONFIG_BT_CTS_HELPER_API
+#undef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 200809L /* To get gmtime_r()'s prototype */
```

Comment:
It would be simpler to put the feature test macros outside the ifdef. Outside the ifdef, it has potentially better test coverage, so it's safer too.

Reply:
Is there a reason for putting it inside the ifdef?

Reply:
Just that it is only needed in that case. It can be outside.

Reply:
I prefer it's outside as then the file is always compatible with the same standard. This reduces complexity.

If it's inside the ifdef, someone could make a change somewhere else in the file, incompatible with `_POSIX_C_SOURCE 200809L`. At best the ifdef is enabled in a CI test and gives a compile error, but even that feedback is unnecessarily late.

Reply:
Changed